### ISSUE TITLE
SSL settings not persisting in compass favorites

### ISSUE THREAD

**Issue by mshedsilegx** [mongodb-js/compass#1880]
Compass version: 1.20.4
OS: Windows 2016 server

1- Before UI restart
![image](https://user-images.githubusercontent.com/32753787/71992983-f37f9b00-31fb-11ea-9f26-52f69e39ae7e.png)

2- After UI restart
![image](https://user-images.githubusercontent.com/32753787/71993045-0abe8880-31fc-11ea-880e-6f10da5de1aa.png)

it is causing a connection timeout everytime when using the favorite after a fresh start.
Clicking again on the favorite while using 'Fill in connection fields individually' link, causes the correct SSL config to load from the saved connection json file

**Comment by imlucas** [comment]
https://jira.mongodb.org/browse/COMPASS-4066

**Comment by antpaw** [comment]
I've the same problem with the Password field, it doesn't get saved in favs

**Comment by Tofandel** [comment]
The ssh tunnel settings as well, so basically none of the fields in `More options` are saved

**Comment by alexndreazevedo** [comment]
All the favorites are saving as a connection string. Even when I set the fields individually it returns to connection string when opened.

**Comment by JasonDepp** [comment]
have the same issue under Compass Version 1.20.5

**Comment by Kadrian** [comment]
I can confirm this under  `1.20.5` and OS X, and it  did work in an older version before, but I'm not sure in which version. How can we help fix this? Does anyone know when / where it was taken out?

**Comment by Anemy** [comment]
Pinging folks that have commented: @Kadrian  @JasonDepp @Tofandel @antpaw @mshedsilegx 
We've just released a new version of Compass, `v1.31.0`, with a number of improvements for connectivity and how we build, save, and load connections. This issue with settings not persisting should be fixed with the changes.
https://www.mongodb.com/try/download/compass
https://github.com/mongodb-js/compass/releases
Feel free to reopen if you update and it's still happening. Sorry for the long turnaround on this, it took us a while to rebuild up this experience to be more reliable and streamlined. Thanks for reporting the issue and trying out Compass!

**Comment by federicobaier** [comment]
I have version 1.32.2 and I'm having the same issue. It refuses to save the favorite if I have SSL configuration in the connection string.

**Comment by Anemy** [comment]
@federicobaier Can you check if you're using any unescaped characters in the connection string? This might be caused by the issue mentioned in: https://jira.mongodb.org/browse/COMPASS-5745 where we fail to load connections with unescaped characters in their connection strings. It's an issue we'd like to get a fix out for sooner.

**Comment by federicobaier** [comment]
No, I don't see any unescaped characters. This is the connection string (without the cluster domain):

`mongodb+srv://here.goes.the.cluster.domain/?authSource=%24external&authMechanism=MONGODB-X509&retryWrites=true&w=majority&tls=true&tlsCertificateKeyFile=%2FUsers%2Ffedebaier%2Fdev%2Fcerts%2Fdb-cert.pem`

The connection works fine, it just doesn't save it as favorite.

**Comment by Anemy** [comment]
@frederickferby Thanks for pinging us on this. We were able to reproduce, looks like https://github.com/mongodb-js/compass/pull/3263 should fix it. We'll have that in the next Compass release.

**Comment by federicobaier** [comment]
Thanks @Anemy! Looking forward for the fix.

**Comment by ingria** [comment]
I'm on 1.38.2 and I have the same issue with password - it disappears from saved connection after compass restart.

**Comment by luigiaimage** [comment]
I'm on version 1.41 and the password is *still* not being saved.

**Comment by jhartzell** [comment]
Also having the same issue under arch linux

**Comment by MattNannetti** [comment]
> I'm on version 1.41 and the password is _still_ not being saved.

Im also having the same issue on 1.42.3 RedHat.

**Comment by ldepaulatrac** [comment]
i found the reason, the XDG_CURRENT_DESKTOP is not set, or its set to something like Hyprland (in my case) so to this work you need to set it as GNOME our other acceptable  you can find the list here

https://source.chromium.org/chromium/chromium/src/+/main:base/nix/xdg_util.cc;l=146-169

and for electron specific (the case of mongo)

https://github.com/electron/electron/issues/32598

https://github.com/electron/electron/issues/32598#issuecomment-1648615204 the comment for the savior


TL/DR: XDG_CURRENT_DESKTOP='GNOME' mongodb-compass fixes

**Comment by mesmerx** [comment]
i found the reason, the XDG_CURRENT_DESKTOP is not set, or its set to something like Hyprland (in my case) so to this work you need to set it as GNOME our other acceptable you can find the list here

https://source.chromium.org/chromium/chromium/src/+/main:base/nix/xdg_util.cc;l=146-169

and for electron specific (the case of mongo)

https://github.com/electron/electron/issues/32598

https://github.com/electron/electron/issues/32598#issuecomment-1648615204 the comment for the savior

TL/DR: XDG_CURRENT_DESKTOP='GNOME' mongodb-compass fixes

**Comment by usersina** [comment]
> i found the reason, the XDG_CURRENT_DESKTOP is not set, or its set to something like Hyprland (in my case) so to this work you need to set it as GNOME or other acceptable that you can find the list here
> 
> https://source.chromium.org/chromium/chromium/src/+/main:base/nix/xdg_util.cc;l=146-169
> 
> and for electron specific (the case of mongo)
> 
> [electron/electron#32598](https://github.com/electron/electron/issues/32598)
> 
> [electron/electron#32598 (comment)](https://github.com/electron/electron/issues/32598#issuecomment-1648615204) the comment for the savior
> 
> you just need the keyring or wallet or any polkit for this passwords to be runing
> 
> TL/DR: `XDG_CURRENT_DESKTOP='GNOME' mongodb-compass ` fixes the issue

For a more permanent solution with a desktop icon, see [this thread](https://askubuntu.com/a/923393/1202781).

**Comment by flopana** [comment]
> i found the reason, the XDG_CURRENT_DESKTOP is not set, or its set to something like Hyprland (in my case) so to this work you need to set it as GNOME or other acceptable that you can find the list here
> 
> https://source.chromium.org/chromium/chromium/src/+/main:base/nix/xdg_util.cc;l=146-169
> 
> and for electron specific (the case of mongo)
> 
> [electron/electron#32598](https://github.com/electron/electron/issues/32598)
> 
> [electron/electron#32598 (comment)](https://github.com/electron/electron/issues/32598#issuecomment-1648615204) the comment for the savior
> 
> you just need the keyring or wallet or any polkit for this passwords to be runing
> 
> TL/DR: `XDG_CURRENT_DESKTOP='GNOME' mongodb-compass ` fixes the issue

I have installed mongodb compass trough flatpak and you can override the behaviour with `flatpak override`

```bash
sudo flatpak override com.mongodb.Compass --env=XDG_CURRENT_DESKTOP='GNOME'
```
